src/sentryFilter/index.test.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import {
+  filterCodeRedirects,
+  containsSensitiveInformation,
+  redactCodeParameter,
+  redactNavigationCrumbs,
+} from "./index";
+
+describe("containsSensitiveInformation", () => {
+  it("returns false when there is no sensitive information given", () => {
+    const string = "hello world";
+    expect(containsSensitiveInformation(string)).toBe(false);
+  });
+
+  it("returns true when there is a code parameter", () => {
+    const string = "code=1a2b3c4d&foo=bar";
+    expect(containsSensitiveInformation(string)).toBe(true);
+  });
+});
+
+describe("redactCodeParameter", () => {
+  it("returns the given string if no code parameter is present", () => {
+    const string = "hello";
+    expect(redactCodeParameter(string)).toEqual(string);
+  });
+
+  it("returns a redacted code parameter when given a code parameter", () => {
+    const string = "/?code=VSDX2EfjlNyZ1fwEiusfU5ZQcPUsODv82twW0fLVs3U&foo=bar";
+    const expected = "/?code=[REDACTED]&foo=bar";
+    expect(redactCodeParameter(string)).toEqual(expected);
+  });
+});
+
+describe("redactNavigationCrumbs", () => {
+  it("does nothing when no sensitive data is present", () => {
+    const crumbs = [{ type: "test", foo: "bar" }];
+    const expectedCrumbs = redactNavigationCrumbs(crumbs);
+    expect(expectedCrumbs[0]).toMatchObject(crumbs[0]);
+  });
+});
+
+describe("filterCodeRedirects", () => {
+  it("does nothing when there is no data to filter", () => {
+    const event = {
+      breadcrumbs: [{ category: "test" }],
+    };
+
+    expect(filterCodeRedirects(event)).toMatchObject(event);
+  });
+
+  it("redacts codes sent in navigation breadcrumbs", () => {
```

Comment:
It'd be a good idea to have a test case that checks multiple breadcrumbs, e.g., navigation, event, etc to ensure non-navigation crumbs aren't redacted/filtered

Reply:
:+1: